compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/diagnostics/coneDiagnosticToFirDiagnostic.kt

```diff
@@ -42,7 +40,6 @@ private fun ConeDiagnostic.toFirDiagnostic(source: FirSourceElement): FirDiagnos
     is ConeWrongNumberOfTypeArgumentsError ->
         FirErrors.WRONG_NUMBER_OF_TYPE_ARGUMENTS.on(source, this.desiredCount, this.type)
     is ConeSimpleDiagnostic -> when {
-        source.kind is FirFakeSourceElementKind -> null
```

Comment:
Without the type, there would be `FirImplicitTypeRef` with a fake source. So, removing this is relevant to get through the factory below.

However, removing this for the whole `ConeSimpleDiagnostic` might be too much or could cause an issue later. Better to be more specific to `VALUE_PARAMETER_WITH_NO_TYPE_ANNOTATION`

Reply:
When I removed it I was thinking it's redundant because the check is already done in org.jetbrains.kotlin.fir.analysis.collectors.AbstractDiagnosticCollector.Visitor#visitTypeRef. But now I think about this again. I think the redundancy is actually a good thing because having it makes the intention explicit and the code more robust.

Reply:
Oh, so that means, it wasn't about going to `factory`?

Reply:
@jsjeon Not sure what you mean. Without the type, the `returnTypeRef` on the value parameter is `FirErrorTypeRef` _and_ there is no source...

Reply:
🤦  Duh, right. Then, again, I think it's needed to get through `factory` and generate the corresponding `FirErrors`?

Reply:
So I was thinking about generating all the FIR diagnostics with fake source inside the `FirFunctionParameterChecker` and leave those that has real source here. If we let the fake through here, we would report duplicated diagnostics (though some of the duplicates, like VALUE_PARAMETER_WITH_NO_TYPE_ANNOTATION would not surface up since there is no source.)

Anyway, if we do want fake source to go through, this change is not enough (as I mentioned above there is another check for fakeness in org.jetbrains.kotlin.fir.analysis.collectors.AbstractDiagnosticCollector.Visitor#visitTypeRef). So for this PR, let me just revert this change.

Reply:
@tgeng Can correct me if I'm wrong but AFACIT L148 (`DiagnosticKind.ValueParameterWithNoTypeAnnotation -> ...`) will never be true. With current code anyway, unless in the future that diagnostic becomes present on something with source.

Reply:
Yes that's right. We could move it to the else case. But I think it's OK to leave it as it is.